internal/websocket/options.go

```diff
@@ -218,16 +178,51 @@ func SendTimeout(d time.Duration) Option {
 		})
 }
 
-// ConnectTimeout sets the timeout for the WS connection.  If this is not set,
-// the default is 30 seconds.
-func ConnectTimeout(d time.Duration) Option {
+// HTTPClient is the configuration for the HTTP client used to retrieve the
+// credentials.
+func HTTPClient(c arrangehttp.ClientConfig) Option {
```

Comment:
This needs to be an optional Option.  Right now it is required since it creates the http.Client used.  I think this is a mistake & that this should largely just take a `*http.Client` in as input.  Later (after the options are handled) the final `*http.Client` should be created given the need for the `ipv4` vs `ipv6` choice as well as which network interface to attempt to use.

Reply:
great point(!), I updated it such that `func HTTPClient(c arrangehttp.ClientConfig)` is actually optional and the http client is created as needed 🙂